### PR TITLE
[CALCITE] Add validation support for host variables

### PR DESCRIPTION
**Pull request by dasch-1**

### CODE REVIEW

File: parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java [hunk 1]
```diff
@@ -41,4 +41,11 @@
         + "FROM `ABC` AS `ABC`";
     sql(sql).rewritesTo(expected);
   }
+
+  @Test public void testHostVariable() {
```
Comment (dstekol) [inline parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java:45]: Are there any restrictions we should be checking regarding where host variables can/cannot occur, what type they have, etc?

Comment (dasch-1) [reply parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java:45]: For now, I'd just like to be able to not alter host variables during validation rewriting. Actually extracting meaningful type information for them would require a larger scope effort (which may or may not be necessary given current needs).

Comment (dasch-1) [reply parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java:45]: The location where host variables can occur should already be enforced during parse time.

Comment (dstekol) [reply parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ValidatorTest.java:45]: Ok, SGTM.

**Pull request merged**